commit cc598989d2d952610a82d556d81d0835f78ef494 (refs/changes/95/395/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-09-01T14:39:16-07:00 (3 years, 1 month ago)
    
    joyent/node-cueball#11 don't create a DNS UDP socket per Resolver

diff --git a/lib/resolver.js b/lib/resolver.js
index ebc004d..9dbb9fa 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -240,9 +240,14 @@ function CueBallDNSResolver(options) {
 	this.r_bootstrap = undefined;
 	this.r_bootstrapRes = {};
 
-	this.r_nsclient = new mod_nsc.DnsClient({
-		concurrency: this.r_maxres
-	});
+	this.r_nsclient = CueBallDNSResolver.globalNSClients[this.r_maxres];
+	if (this.r_nsclient === undefined) {
+		this.r_nsclient = new mod_nsc.DnsClient({
+			concurrency: this.r_maxres
+		});
+		CueBallDNSResolver.globalNSClients[this.r_maxres] =
+		    this.r_nsclient;
+	}
 
 	this.r_stopping = false;
 
@@ -254,6 +259,8 @@ mod_util.inherits(CueBallDNSResolver, FSM);
 
 CueBallDNSResolver.bootstrapResolvers = {};
 
+CueBallDNSResolver.globalNSClients = {};
+
 CueBallDNSResolver.prototype.start = function () {
 	this.emit('startAsserted');
 };
